Assets/Scripts/Engine/Engine.cs

```diff
@@ -95,12 +104,20 @@ public bool isKeyDown()
         }
 	}
 
+	public bool isKeyUp()
+	{
+		return Input.GetKeyUp (KeyCode.Space) || Input.GetMouseButtonUp (0);
+	}
+
 	public bool isPauseKeyDown()
 	{
 		return Input.GetKeyDown (KeyCode.Escape);
 	}
 
-
+	public bool getKeyPressed()
+	{
+		return keyPressed;
```

Comment:
You can just use Input.GetKey. You don't need to use the update function and boolean stuff.